starter-code/routes/profile.js

```diff
@@ -0,0 +1,79 @@
+/*jshint esversion: 6 */
+
+const express = require('express');
+const router = express.Router();
+const User = require("../models/user");
+const bcrypt  = require("bcrypt");
+const saltRounds   = 10;
+
+
+// Edit the User
+router.get('/:userId/edit', (req, res, next) => {
+  const userId = req.params.userId;
+  User.findById(userId, (err, user) => {
+    if (!req.session.user) {
+      res.redirect('/login');
+    } else if (user.username === req.session.user.username) {
+        res.render('profiles/edit', {
+          user: user,
+        });
+    } else {
+        return next(err);
+    }
+  });
+});
+
+// Info to update the user
+router.post('/:userId', (req, res, next) => {
+  const userId = req.params.userId;
+
+  let updates = {
+    name: req.body.name,
+    email: req.body.email,
+    imageUrl: req.body.imageUrl,
+    summary: req.body.summary,
+    company: req.body.company,
+    jobTitle: req.body.jobTitle,
+    username: req.body.username
+  };
+
+  // User.findOne({"username" : req.body.username}, "username", (err, user) => {
```

Comment:
why did you comment this out?

you should keep this check, and move the remaining code into the callback on an `else` block